addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/AuthUtils.java

```diff
@@ -533,6 +555,31 @@ static User getUser(Context context, String userName) {
         return oUser.get();
     }
 
+    public static VerificationRequestDetails getVerificationDetailsForContext(int contextId) {
+        return contextVerifMap.get(contextId);
+    }
+
+    public static void setVerificationDetailsForContext(
+            int contextId, VerificationRequestDetails details) {
+        contextVerifMap.put(contextId, details);
+    }
+
+    private static ExecutorService executorService;
```

Comment:
- I'd suggest moving to the top, we usually keep them all together there.
- Needs to be stopped at some point (extension unload?), e.g. in command line mode it could prevent ZAP from exiting as the executor uses non-daemon thread.